src/settings/SettingsSelfExclusion.js

```diff
@@ -88,6 +91,7 @@ export default class SettingsSelfExclusion extends PureComponent {
         success: true,
         serverError: false,
       });
+      document.getElementsByClassName('update-notice')[0].scrollIntoView({ block: 'center', behavior: 'smooth' });
```

Comment:
@arnabk @aminroosta guys. I have here a preview of using refs instead of getElementById: https://github.com/brucebinary/binary-next-gen/compare/scroll-to-error-fix...brucebinary:scroll-to-error-finddomnode My point is to demonstrate that using refs in this context is tedious and difficult to understand, for something that is seemingly trivial. What is happening is that the we are still accessing the DOM node, but instead of doing it directly (react components don't have a `scrollIntoView` method, unless you install a library for it) via `document` we access it through the corresponding react component via `findDOMNode` (use of `findDOMNode` is actually ill-advised and eslint will throw errors for it). 

However, if everyone agrees to do it this way I will update this PR accordingly.

However I'm a little stuck with the above line of code. What it does it scroll to the update notice when the settings have successfully updated. Should we decide to use refs, the ref to this `UpdateNotice` component will need to be passed to this component, in which we again use it to find the corresponding DOM node via `findDOMNode` and call `scrollIntoView`.

Bear in mind all we want to do is to scroll the view to a specified input field.

Reply:
@brucebinary Couple of comments

```
const addToFormDOM = inp => { if (inp) this.DOMNodes[inp.props.id] = ReactDOM.findDOMNode(inp); };
```
It might be better not to have this function inside render method.
Try ` if (inp) this.DOMNodes[inp.props.id] = inp;`. Not sure if it will work, but worth a try.

https://github.com/stipsan/scroll-into-view-if-needed
Is it at all useful to use this library? Not sure if it would help us in future to scroll differently with different types of animation.

Just a thought. Other than that, I think these changes are as per React patterns, although I do hear your point that just to do the pattern, we are writing double the code.

Reply:
@arnabk the code `if (inp) this.DOMNodes[inp.props.id] = inp;` will not work because `inp` is a react component. What we want is the HTML DOM node.

The scroll library linked also requires the HTML DOM to be passed as argument. It doesn't take a react component. The library I was referring to is this: https://github.com/flyingant/react-scroll-to-component 

We will still need that `addToFormDOM` helper method, but at least we won't use `React.findDOMNode`.